### ISSUE TITLE
Show goal dumps and full anchor specs in verifier errors [phab:D1346]

### ISSUE THREAD

**Issue by kythe** [google/kythe#1740]
_Created by **zarko** at 2017-02-17 22:26:17_: 

When a verifier goal fails, it will now dump
the goal's elaborated form. This makes it easier
to debug anchor-related issues.

Diagnostics will also show the full location specification
when an anchor's goals fail to unify.

Example:
```
Could not verify all goals. The furthest we reached was:
  kythe/cxx/indexer/cxx/testdata/basic/typeof_param.c:2:5-2:7 x
  Goal: App(=, (Range(128,131), EVar(0x000000000302f600 = <nullptr>)))
```

**Comment by kythe** [comment]
_**shahms** accepted this revision at 2017-02-17 22:27:09_

**Comment by kythe** [comment]
_**zarko** closed via commit 3dc7a00914932c82fc4d98f46896cad90767dfd5 at 2017-02-17 22:28:02_
